pkg/telemetry/simple_counter.go

```diff
@@ -0,0 +1,33 @@
+package telemetry
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// SimpleCounter tracks how many times something is happening.
+type SimpleCounter interface {
+	// Inc increments the counter.
+	Inc()
+	// Add increments the counter by given amount.
+	Add(float64)
+}
+
+// NewSimpleCounter creates a new SimpleCounter with default options.
+func NewSimpleCounter(subsystem, name, help string) SimpleCounter {
```

Comment:
It seems to not be used.

Reply:
My understanding was that the interface `SimpleCounter` was used to not expose the return type of [WithValues](https://github.com/DataDog/datadog-agent/pull/8756/files/865c02eba9aeaad01a8f3ca9c733e7855fbe0680#diff-27f89cdbfd2dea42cc343309bd42a99714255410d48269b1cc03b9cd351b244fR37). In this case having a constructor is not common.

Reply:
It is not only to hide the return type, `SimpleCounter` can be used on its own.